Use fetch API

running-on-sunshine opened the pull request:
In this branch: 

- Refactored the following functions using jQuery AJAX with built-in fetch API and promises:
  - retrieveOrders()
  - addOrder()
  - removeOrder()

@clinturbin: Could you please help review? 🙂

running-on-sunshine committed:
Refactored retrieve orders with fetch API & promises

running-on-sunshine committed:
Refactored add orders function with fetch API & promises

running-on-sunshine committed:
Refactored remove orders function with fetch API & promises

clinturbin commented on scripts/main.js:
```diff
@@ -4,35 +4,37 @@ var orders = [];
         }
+    }).then(function () {
+        retrieveOrders();
     })
```
semicolon

clinturbin commented on scripts/main.js:
```diff
@@ -4,35 +4,37 @@ var orders = [];
+        orders = []
+        var ordersToJson = data.json();
+        ordersToJson.then(function (data) {
             for (key in data) {
```
This is a possible refactor: (including semicolons)
```
var retrieveOrders = function () {
  orders = [];
  fetch(url).then(function (data) {
    data.json().then(function (data) {
      for (key in data) {
        orders.push(data[key]);
      })
      displayOrders();
    });
  });
};
```

running-on-sunshine replied:
Thank you for your suggestion! This really helped to clean up and shorten my retrieveOrders function! 🙂

running-on-sunshine committed:
Updated and shortened retrieve orders function

running-on-sunshine commented:
Update:

- Cleaned up retrieveOrders function as a direct result from refactoring suggestion
- Added missing semicolons 😄

@clinturbin: Could you please help review? 🙂

clinturbin commented:
Looks Good!! 👍
